file/random_access_file_reader.cc

```diff
@@ -21,7 +21,10 @@
 
 namespace ROCKSDB_NAMESPACE {
 Status RandomAccessFileReader::Read(uint64_t offset, size_t n, Slice* result,
-                                    char* scratch, bool for_compaction) const {
+                                    char* scratch,
+                                    std::unique_ptr<const char[]>* internal_buf,
```

Comment:
I put them together because I thought it's related to `scratch`. Thanks for pointing this out.

Reply:
I still prefer to put `internal_buf` before `for_compaction` because:
1. the purpose of `internal_buf` is similar to that of `result` and `scratch`
2. if `internal_buf` is put after `for_compaction`, then either we provide a default value for `internal_buf`(I don't want to do this because then callers will by default always copy to scratch in direct IO mode, I want to make it explicit for the caller to choose the copy behavior) or we cannot use default value for `for_compaction` (which is inconvenient).